zerver/lib/import_realm.py

```diff
@@ -1328,21 +1328,24 @@ def do_import_realm(import_dir: Path, subdomain: str, processes: int = 1) -> Rea
     bulk_import_model(data, Reaction)
 
     # Similarly, we need to recalculate the first_message_id for stream objects.
-    update_first_message_id_query = """
+    update_first_message_id_query = SQL(
+        """
     UPDATE zerver_stream
     SET first_message_id = subquery.first_message_id
     FROM (
         SELECT r.type_id id, min(m.id) first_message_id
         FROM zerver_message m
         JOIN zerver_recipient r ON
         r.id = m.recipient_id
-        WHERE r.type = 2
+        WHERE r.type = 2 AND m.realm_id = %(realm_id)s
         GROUP BY r.type_id
         ) AS subquery
     WHERE zerver_stream.id = subquery.id
```

Comment:
I'm not sure if the UPDATE itself should also be constrained to the realm?

Reply:
Good question.  I don't _think_ it's necessary -- it's not going to speed up the UPDATE, which is going to use the primary key for its lookups, and unless we have database corruption it's not going to limit which rows are updated.